[2.16 - ACCUEIL] Erreur backend sur la route /validation en l'absence de dataset.dataset_name dans la config Frontend

## Version

2.16.4 - L'erreur ne se produit pas en 2.15.4 avec la même configuration.

## Description du bug

Lors de l'ouverture de geonature ou de retour sur la page d'accueil une erreur 500 s'affiche (popup alerte rouge)

<img width="499" height="239" alt="Image" src="https://github.com/user-attachments/assets/c471ee7e-383e-4133-81e8-c1f988c31f43" />

## Comment reproduire le bug

Le bug se produit si la colonne `jeu de données` a été retiré de la liste des colonnes dans la config du module validation.
Pour reproduire, faire un fichier `~/geonature/config/validation_config.toml` sans la partie ci-dessous

```
[[COLUMN_LIST]]
column_label = "Jeu de données"
column_name = "dataset.dataset_name"
max_width = 100
```

## Logs

/var/log/geonature/geonature.log : 

```
[2026-02-01 14:03:50 +0100] [1811939] [ERROR] Exception on /validation [GET]
Traceback (most recent call last):
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/home/geonature/geonature/backend/geonature/core/gn_permissions/decorators.py", line 83, in decorated_view
    return view_func(*args, **kwargs)
  File "/home/geonature/geonature/contrib/gn_module_validation/backend/gn_module_validation/blueprint.py", line 173, in get_synthese_data
    dataset_index = relationships.index("dataset")
ValueError: 'dataset' is not in list
```

En console 
```
ERROR 
Object { headers: {…}, status: 500, statusText: "Internal Server Error", url: "https://mondomaine.fr/geonature/api/validation?page=1&per_page=10&sort=desc&order_by=last_validation.validation_date&format=json&fields=id_synthese,nom_cite,observers,date_min,date_max,last_validation,nomenclature_valid_status.cd_nomenclature,nomenclature_valid_status.mnemonique,nomenclature_valid_status.label_default,validator&no_auto=true", ok: false, name: "HttpErrorResponse", message: "Http failure response for https://mondomaine.fr/geonature/api/validation?page=1&per_page=10&sort=desc&order_by=last_validation.validation_date&format=json&fields=id_synthese,nom_cite,observers,date_min,date_max,last_validation,nomenclature_valid_status.cd_nomenclature,nomenclature_valid_status.mnemonique,nomenclature_valid_status.label_default,validator&no_auto=true: 500 Internal Server Error", error: {…} }
​
error: Object { code: 500, description: "The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.", name: "Internal Server Error", … }
​
headers: Object { normalizedNames: Map(0), lazyUpdate: null, lazyInit: lazyInit() }
​
message: "Http failure response for https://mondomaine.fr/geonature/api/validation?page=1&per_page=10&sort=desc&order_by=last_validation.validation_date&format=json&fields=id_synthese,nom_cite,observers,date_min,date_max,last_validation,nomenclature_valid_status.cd_nomenclature,nomenclature_valid_status.mnemonique,nomenclature_valid_status.label_default,validator&no_auto=true: 500 Internal Server Error"
​
name: "HttpErrorResponse"
​
ok: false
​
status: 500
​
statusText: "Internal Server Error"
​
url: "https://mondomaine.fr/geonature/api/validation?page=1&per_page=10&sort=desc&order_by=last_validation.validation_date&format=json&fields=id_synthese,nom_cite,observers,date_min,date_max,last_validation,nomenclature_valid_status.cd_nomenclature,nomenclature_valid_status.mnemonique,nomenclature_valid_status.label_default,validator&no_auto=true"
​
<prototype>: Object { … }
main.99703720bf2aa423.js:1:1515107

```

--- a/backend/geonature/tests/test_validation.py
+++ b/backend/geonature/tests/test_validation.py
@@ -24,6 +24,7 @@ pytestmark = pytest.mark.skipif(
 )
 
 from gn_module_validation.tasks import set_auto_validation
+from gn_module_validation.constant import DEFAULT_FIELDS, DEFAULT_PROFILE_FIELDS
 
 
 @pytest.fixture()
@@ -362,6 +363,21 @@ class TestValidationRoutes:
     # ============================================================================
     # GET /observations - Tests for get_observations_last_validations
     # ============================================================================
+
+    def test_get_observations_with_fields_parameter(self, users, synthese_data):
+        set_logged_user(self.client, users["user"])
+        response = self.client.get(
+            url_for("validation.get_observations_last_validations", fields="nomenclature_blurring")
+        )
+        assert response.status_code == 200
+        assert "features" in response.json
+        data = response.json["features"][0]["properties"]
+        for field in DEFAULT_FIELDS:
+            assert field.split(".")[0] in data
+        for field in DEFAULT_PROFILE_FIELDS:
+            assert field.split(".")[0] in data
+        assert "nomenclature_blurring" in data
+        assert response.status_code == 200
 
     def test_get_observations_unauthorized(self):
         """Test that unauthorized access is rejected."""

--- a/contrib/gn_module_validation/backend/gn_module_validation/blueprint.py
+++ b/contrib/gn_module_validation/backend/gn_module_validation/blueprint.py
@@ -91,7 +91,6 @@ def get_observations_last_validations(permissions):
 
     # Execute query
     query = query_statement.from_statement(selectable)
-
     return jsonify(query.as_geofeaturecollection(fields=fields))
 
 

--- a/contrib/gn_module_validation/backend/gn_module_validation/query_build.py
+++ b/contrib/gn_module_validation/backend/gn_module_validation/query_build.py
@@ -40,16 +40,14 @@ def get_fields_from_params(params: Dict) -> Set[str]:
     enable_profile = current_app.config["FRONTEND"]["ENABLE_PROFILES"]
     fields_str = params.pop("fields", None)
 
-    if fields_str:
-        fields = {field.strip() for field in fields_str.split(",")}
-    else:
-        fields = DEFAULT_FIELDS.copy()
-        if enable_profile:
-            fields.update(DEFAULT_PROFILE_FIELDS)
+    fields = set(list(DEFAULT_FIELDS))
+    if enable_profile:
+        fields.update(DEFAULT_PROFILE_FIELDS)
 
+    if fields_str:
+        fields.update({field.strip() for field in fields_str.split(",")})
     # Add configured columns
     fields.update({col["column_name"] for col in current_app.config["VALIDATION"]["COLUMN_LIST"]})
-
     return fields
 
 

